### PR TITLE
pybind/mgr/rbd_support: delay creation of progress module events

### PR DESCRIPTION
**Pull request by dillaman**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/rbd_support/task.py [hunk 1]
```diff
@@ -373,7 +374,6 @@ def execute_task(self, sequence):
                 else:
                     task.in_progress = True
                     self.in_progress_task = task
-                    self.update_progress(task, 0)
```
Comment (trociny) [inline src/pybind/mgr/rbd_support/task.py:367]: @dillaman Don't you think we can also remove `update_progress` when handling exception?

**Review by trociny — APPROVED** [review]
LGTM

**Comment by trociny** [comment]
Just an observation, when I was testing it and a task was running, the `ceph -s` was printing 'progress:' string but showing no tasks (progress bar). I have no idea if it is expected behavior, and suppose it is not related to this PR.

**Comment by dillaman** [comment]
> I have no idea if it is expected behavior, and suppose it is not related to this PR.

Yeah, I noticed the same thing before I starting changing things. It's still visible via 'ceph progress' though.

File: src/pybind/mgr/rbd_support/task.py (hunk 1 above)
Comment (dillaman) [reply src/pybind/mgr/rbd_support/task.py:367]: I was figuring that if we hit an exception after the task started I would want to know the reason (so I added the `retry_message`). The `update_progress` on the exception path won't create a new progress event unless one was already posted via the librbd progress callback.

**Pull request merged**